pkg/xds/envoy/clusters/v2/health_check_configurer.go

```diff
@@ -94,6 +97,39 @@ func httpHealthCheck(
 	}
 }
 
+func healthPanicThreshold(cluster *envoy_api.Cluster, value *wrappers.FloatValue) {
+	if value == nil {
+		return
+	}
+	if cluster.CommonLbConfig == nil {
+		cluster.CommonLbConfig = &envoy_api.Cluster_CommonLbConfig{}
+	}
+	cluster.CommonLbConfig.HealthyPanicThreshold = &envoy_type.Percent{Value: float64(value.Value)}
+}
+
+func failTrafficOnPanic(cluster *envoy_api.Cluster, value *wrappers.BoolValue) {
+	if value == nil {
+		return
+	}
+	if cluster.CommonLbConfig == nil {
+		cluster.CommonLbConfig = &envoy_api.Cluster_CommonLbConfig{}
+	}
+	if cluster.CommonLbConfig.GetLocalityWeightedLbConfig() != nil {
+		// used load balancing type doesn't support 'fail_traffic_on_panic', right now we don't use
+		// 'locality_weighted_lb_config' in Kuma, locality aware load balancing is implemented based on priority levels
+		core.Log.WithName("health-check-configurer").Error(
+			errors.New("unable to set 'fail_traffic_on_panic' for 'locality_weighted_lb_config' load balancer"),
+			"unable to configure 'fail_traffic_on_panic', parameter is ignored")
+		return
+	}
+	if cluster.CommonLbConfig.LocalityConfigSpecifier == nil {
+		cluster.CommonLbConfig.LocalityConfigSpecifier = &envoy_api.Cluster_CommonLbConfig_ZoneAwareLbConfig_{
+			ZoneAwareLbConfig: &envoy_api.Cluster_CommonLbConfig_ZoneAwareLbConfig{},
+		}
+	}
+	cluster.CommonLbConfig.GetZoneAwareLbConfig().FailTrafficOnPanic = value.GetValue()
```

Comment:
So is this the only other option if `cluster.CommonLbConfig.GetLocalityWeightedLbConfig()` is empty?

Reply:
Yes, this is a good point, now there are only 2 options, but it might change. So I added nil-check, thanks!